src/rust/engine/src/externs/mod.rs

```diff
@@ -51,57 +47,51 @@ pub fn store_tuple(py: Python, values: Vec<Value>) -> Value {
     .into_iter()
     .map(|v| v.consume_into_py_object(py))
     .collect();
-  Value::from(PyTuple::new(py, &arg_handles).into_object())
+  Value::from(PyTuple::new(py, &arg_handles).to_object(py))
 }
 
 /// Store a slice containing 2-tuples of (key, value) as a Python dictionary.
-pub fn store_dict(py: Python, keys_and_values: Vec<(Value, Value)>) -> Result<Value, PyErr> {
+pub fn store_dict(py: Python, keys_and_values: Vec<(Value, Value)>) -> PyResult<Value> {
   let dict = PyDict::new(py);
   for (k, v) in keys_and_values {
-    dict.set_item(
-      py,
-      k.consume_into_py_object(py),
-      v.consume_into_py_object(py),
-    )?;
+    dict.set_item(k.consume_into_py_object(py), v.consume_into_py_object(py))?;
   }
-  Ok(Value::from(dict.into_object()))
+  Ok(Value::from(dict.to_object(py)))
 }
 
 /// Store an opaque buffer of bytes to pass to Python. This will end up as a Python `bytes`.
 pub fn store_bytes(py: Python, bytes: &[u8]) -> Value {
-  Value::from(PyBytes::new(py, bytes).into_object())
+  Value::from(PyBytes::new(py, bytes).to_object(py))
 }
 
-/// Store an buffer of utf8 bytes to pass to Python. This will end up as a Python `str`.
+/// Store a buffer of utf8 bytes to pass to Python. This will end up as a Python `str`.
 pub fn store_utf8(py: Python, utf8: &str) -> Value {
-  Value::from(utf8.to_py_object(py).into_object())
+  Value::from(utf8.to_object(py))
 }
 
 pub fn store_u64(py: Python, val: u64) -> Value {
-  Value::from(val.to_py_object(py).into_object())
+  Value::from(val.to_object(py))
 }
 
 pub fn store_i64(py: Python, val: i64) -> Value {
-  Value::from(val.to_py_object(py).into_object())
+  Value::from(val.to_object(py))
 }
 
 pub fn store_bool(py: Python, val: bool) -> Value {
-  Value::from(val.to_py_object(py).into_object())
+  Value::from(val.to_object(py))
 }
 
 ///
 /// Gets an attribute of the given value as the given type.
 ///
-pub fn getattr<T>(value: &PyObject, field: &str) -> Result<T, String>
+pub fn getattr<'py, T>(value: &'py PyAny, field: &str) -> Result<T, String>
 where
-  for<'a> T: FromPyObject<'a>,
+  T: FromPyObject<'py>,
```

Comment:
Note the lifetime change. I didn't fully get how the original was working, but I'm pretty sure this new one is right. The `'py` corresponds to the lifetime of the GIL / `Python` token.

Reply:
Yea, makes sense. Before, `T` was not going to have its own lifetime (it was implicitly `'static`), but now it does.